### PR TITLE
*: support get disk capacity and available.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @disksing @tiancaiamao

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -714,18 +715,40 @@ impl<T: Transport, C: PdClient> Store<T, C> {
...
+                error!("get disk stat for rocksdb {} failed", self.engine.path());
+                return;
+            }
+        };
+
+        let capacity = if disk_stat.capacity > self.cfg.capacity {
+            disk_stat.capacity
+        } else {
+            self.cfg.capacity
+        };
```
Comment (disksing) [inline src/raftstore/store/store.rs]: Swap L727 and L729?

**Comment by disksing** [comment]
LGTM

File: src/raftstore/store/store.rs [hunk 2]
```diff
@@ -714,18 +716,36 @@ impl<T: Transport, C: PdClient> Store<T, C> {
...
-        stats.set_capacity(self.cfg.capacity);
+        let disk_stat = match get_disk_stat(self.engine.path()) {
+            Ok(disk_stat) => disk_stat,
+            Err(_) => {
+                error!("get disk stat for rocksdb {} failed", self.engine.path());
+                return;
+            }
+        };
+
+        let capacity = cmp::min(disk_stat.capacity, self.cfg.capacity);
```
Comment (BusyJay) [inline src/raftstore/store/store.rs]: What if `cfg.capacity` is 0?

File: src/util/fs.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+// Get the disk stats for path belongs.
+// TODO: define own Error type instead of string.
+pub fn get_disk_stat(path: &str) -> Result<DiskStat, String> {
+    let cpath = CString::new(path).unwrap();
+    unsafe {
+        let mut stat: libc::statfs = mem::zeroed();
+        let ret = libc::statfs(cpath.as_ptr(), &mut stat);
+        if ret != 0 {
+            return Err(format!("get stats for {} failed", path));
```
Comment (BusyJay) [inline src/util/fs.rs]: I think it would be better to format the error code too.

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**